### PR TITLE
Wave 2

### PR DESCRIPTION
**Pull request by laerreal**
Yet more patches from `devel`.
Related patches are not squashed but placed near each other.

### v2
- some new patches from `devel`
- reorder patches
- missed `except StopVisiting`
- fixing commits inserted
- fix spacing in `TkGeometryHelper` related commits

### CODE REVIEW

File: common/visitor.py [hunk 1]
```diff
@@ -185,17 +201,29 @@ def __visit__(self, attr):
...
+                raise
             self.__pop__()
 
     def __visit_dictionary__(self, attr):
         for k, e in sorted(attr.items()):
             self.__push__(e, k)
-            self.__visit__(e)
+            try:
+                self.__visit__(e)
+            except:
```
Comment (BDanAnd) [inline common/visitor.py]: Why not `except StopVisiting:` only?

File: widgets/var_widgets.py [hunk 2]
```diff
@@ -273,6 +281,27 @@ def heading(self, column, **kw):
...
+
+        iid = Treeview.insert(self, *a, **kw)
+
+        for col, v in to_track:
+            TreeviewCellBinding(self, col, iid, v)
+
+        return iid
+
+    def item(self, iid, *a, **kw):
+
```
Comment (BDanAnd) [inline widgets/var_widgets.py]: An extra line spoils the beauty.

**Review by BDanAnd — COMMENTED** [review]
v1 viewed

**Comment by laerreal** [comment]
v2 is ready

**Pull request merged**